test/unit/widgets/forms.test.ts

```diff
@@ -1,1435 +1,1455 @@
+        <td
+          class="formFieldName"
+          style="  vertical-align: middle;"
+        >
+          <a
+            href="http://example.com/#bla"
+            style="color: #3B5998; text-decoration: none;"
+          >
+            Bla
+          </a>
+        </td>
+        <td
+          class="formFieldValue"
+        >
+          <input
+            maxlength="4096"
+            size="20"
+            style="background-color: #eef; padding: 0.5em;  border: .05em solid #88c;  border-radius:0.2em; font-size: 100%; margin:0.2em; "
+            type="text"
+          />
+        </td>
+      </tr>
+    `)
+  })
+})
+
+describe('NamedNodeURIField', () => {
+  it('exists', () => {
+    expect(field[ns.ui('NamedNodeURIField').uri]).toBeInstanceOf(Object)
+  })
+  it('runs', () => {
+    const container = document.createElement('div')
+    const already = {}
+    const subject = namedNode('http://example.com/#this')
+    const form = namedNode('http://example.com/#form')
+    const store = namedNode('http://example.com/#store')
+    const callbackFunction = () => {}
+    uiStore.add(form, ns.rdf('type'), namedNode('http://example.com/#type'), namedNode('http://example.com/'))
+    uiStore.add(form, ns.ui('property'), namedNode('http://example.com/#bla'), namedNode('http://example.com/'))
+
+    expect(
+      field[ns.ui('NamedNodeURIField').uri](
+        document,
+        container,
+        already,
+        subject,
+        form,
+        store,
+        callbackFunction
+      )
+    ).toMatchInlineSnapshot(`
+      <tr>
+        <td
+          class="formFieldName"
+          style="  vertical-align: middle;"
+        >
+          <a
+            href="http://example.com/#bla"
+            style="color: #3B5998; text-decoration: none;"
+          >
+            Bla
+          </a>
+        </td>
+        <td
+          class="formFieldValue"
+        >
+          <input
+            maxlength="4096"
+            size="20"
+            style="background-color: #eef; padding: 0.5em;  border: .05em solid #88c;  border-radius:0.2em; font-size: 100%; margin:0.2em; "
+            type="text"
+          />
+        </td>
+      </tr>
+    `)
+  })
+})
+
+describe('MultiLineTextField', () => {
+  it('exists', () => {
+    expect(field[ns.ui('MultiLineTextField').uri]).toBeInstanceOf(Object)
+  })
+  it.skip('runs', () => {
+    const container = document.createElement('div')
+    const already = {}
+    const subject = namedNode('http://example.com/#this')
+    const form = namedNode('http://example.com/#form')
+    const store = namedNode('http://example.com/#store')
+    const callbackFunction = () => {}
+    expect(
+      field[ns.ui('MultiLineTextField').uri](
+        document,
+        container,
+        already,
+        subject,
+        form,
+        store,
+        callbackFunction
+      )
+    ).toMatchInlineSnapshot(`
+<div>
+        <a
+          href="http://example.com/#bla"
+          style="color: #3B5998; text-
```

Comment:
Would be good to have a test when this callback is called. Similar with all the other tests that simply give a `() => {}` in the callback parameter.

You probably want to split the test into two tests, one to check the instance, and another to check that the callback was called when expected. Use `beforeEach` to group test code that is called both places.

Reply:
See https://github.com/solid/solid-ui/pull/255#discussion_r392251757